modules/class-simplechart-plugin-versions.php

```diff
@@ -0,0 +1,145 @@
+			$this->_latest_plugin_update_name = __( 'Latest Update', 'simplechart' );
+		}
+
+		if ( false === ( $this->_latest_plugin_update_description = get_transient( 'simplechart_plugin_update_body' ) ) ) {
+			$this->_latest_plugin_update_description = __( 'See more on Github.', 'simplechart' );
+		}
+
+		$this->_latest_plugin_last_updated = get_transient( 'simplechart_plugin_update_last_updated' );
+
+		if ( version_compare( Simplechart::instance()->get_config( 'version' ), $this->_latest_plugin_version, '<' ) ) {
+			add_filter( 'site_transient_update_plugins', array( $this, 'simplechart_extend_filter_update_plugins' ) );
+		}
+	}
+
+	/**
+	 * Add Simplechart to the plugin updates transient when appropraite
+	 * @param object          $update_plugins Contains plugin update info
+	 */
+	public function simplechart_extend_filter_update_plugins( $update_plugins ) {
+		if ( ! is_object( $update_plugins ) ) {
+			return $update_plugins;
+		}
+
+		if ( ! isset( $update_plugins->response ) || ! is_array( $update_plugins->response ) ) {
+			$update_plugins->response = array();
+		}
+
+		$update_plugins->response['wordpress-simplechart/simplechart.php'] = (object) array(
+			'slug'         => 'wordpress-simplechart',
+			'new_version'  => $this->_latest_plugin_version,
+			'url'          => $this->_simplechart_url,
+		);
+		return $update_plugins;
+	}
+
+	/**
+	 * Intercept the plugins API result and return Simplechart remote information
+	 * @param object|WP_Error $res    Response object or WP_Error.
+	 * @param string          $action The type of information being requested from the Plugin Install API.
+	 * @param object          $args   Plugin API arguments.
+	 */
+	public function simplechart_mock_plugins_api( $res, $action, $args ) {
+		if ( 'plugin-information' !== $action && 'wordpress-simplechart' !== $args->slug ) {
+			return $res;
+		}
+		$copy_generated = sprintf(
+			__( 'Ask an administrator to replace your existing Simplechart plugin directory with <a href="%1$s">the ZIP of %2$s via Github</a>', 'simplechart' ),
+			esc_url( $this->_latest_plugin_zip_url ),
+			$this->_latest_plugin_version
+		);
+		$description =
+			'<small>'
+			. $copy_generated
+			. '</small>'
+			. '<h1>'
+			. $this->_latest_plugin_update_name
+			. '</h1>'
+			. $this->_latest_plugin_update_description;
+
+		return (object) array(
+			'name'         => __( 'Simplechart', 'simplechart' ),
+			'banners'      => array(
+				'high'       => null,
+				'low'        => null,
+			),
+			'external'     => $this->_simplechart_url,
+			'homepage'     => $this->_simplechart_url,
+			'slug'         => 'wordpress-simplechart',
+			'version'      => $this->_latest_plugin_version,
+			'last_updated' => $this->_latest_plugin_last_updated,
+			'sections'     => array(
+				'changelog'  => $description,
+			),
+		);
+	}
+
+	/**
+	 * Update the transients for Simple Chart remote version
+	 */
+	public function update_simplechart_remote_version() {
```

Comment:
Maybe rename this method to clarify that you're not only updating `simplechart_plugin_version_remote`. `update_remote_version_metadata()`?